src/Engine/Renderer/Apis/OpenGL/Shader/OpenGLShaderData.cpp

```diff
@@ -152,10 +152,8 @@ namespace gp1::renderer::apis::opengl::shader
 			glGetShaderInfoLog(shaderID, 1024, &length, infoLog);
 			OpenGLShaderData::s_Logger.LogError("%s at %s compile failed:\n%s", GetShaderTypeName(type), ("Shaders/" + shader->GetId() + GetShaderTypeExtensionName(type)).c_str(), infoLog);
```

Comment:
This is where the segfault is coming from. The logger is throwing it here.

The problem is that MSVC allows you to reuse the `va_list` in the logger. That's not actually allowed, and only works on MSVC because they love to, um, "extend" the standard. The shader is failing to compile, and when we go to log that, it blows up.

Thankfully, this is easily (band aid) fixable. Just copy the va_list before you use it the first time, and use the copy once and the original once.

```
va_list args_copy
va_copy(args_copy, args)
...
vsnprintf(buf, length, format, args_copy);
```
In src/Engine/Utility/Logger.cpp.

Reply:
Btw, the reason the shaders are failing to compile and link is because of the paths. So we can't load any of the shaders. Side note, if we fail to compile a shader, it seems to just return 0 and keep going as if everything is fine, no logging, no error handling.